charmcraft/commands/store/client.py

```diff
@@ -47,6 +47,34 @@ def build_user_agent():
     )
 
 
+class AnonymizedClient(craft_store.http_client.HTTPClient):
+    """Lightweight layer that access public store data."""
+
+    def __init__(self, api_base_url, storage_base_url, ephemeral=False):
+        self.api_base_url = api_base_url.rstrip("/")
+        self.storage_base_url = storage_base_url.rstrip("/")
+        del ephemeral  # Placeholder
+
+        super().__init__(
```

Comment:
I'm thinking now that maybe this class should contain an `HTTPClient` instance, as opposed to being one. I say this because now this `AnonymizedClient` has a bunch of public methods like 'get' and 'post' that shouldn't be used directly; the public api should only be those two `request_urlpath_text|json` right?
@facundobatista what do you think?